src/v/storage/ntp_config.h

```diff
@@ -75,11 +75,27 @@ class ntp_config {
       , _overrides(std::move(overrides))
       , _revision_id(id) {}
 
+    ntp_config(
+      model::ntp n,
+      ss::sstring base_dir,
+      std::unique_ptr<default_overrides> overrides,
+      model::revision_id id,
+      model::initial_revision_id initial_id) noexcept
```

Comment:
do we need `intial_revision_id` to be added to `ntp_config` ? Maybe it would be enough for it to be accessible via `cluster::topic_table_api` ? Initial revision doesn't change anything for the `ntp` itself. Adding the initial revision to ntp_config suggest that it actually control ntp behavior.

Reply:
Initially I did just that. But that required to add extra parameter to a lot of constructors and methods (partition, archival_metadata_stm, partition_manager, recovery_manager, archival/service, ntp_archival_service, etc) + some of the classes need an extra field to store initial revision. It was a lot of scattered changes so I decided to add it to ntp_config. The ntp_config is not copied around so one extra field shouldn't consume too much memory. And it affects the way the ntp is interacting with the shadow indexing.